src/ansi-c/c_typecheck_expr.cpp

```diff
@@ -1443,8 +1439,8 @@ void c_typecheck_baset::typecheck_expr_ptrmember(exprt &expr)
 
   // turn x->y into (*x).y
 
-  exprt deref(ID_dereference);
-  deref.move_to_operands(expr.op0());
+  dereference_exprt deref;
```

Comment:
`dereference_exprt deref{expr.op0()}`?

Reply:
Or perhaps since this doesn't actually set the type, perhaps `nil_typet` or something to precisely mimic existing behaviour?